torch/csrc/jit/passes/lower_tuples.cpp

```diff
@@ -32,26 +70,12 @@ static void VisitNode(Node* n, Node* insert_point) {
     return;
   }
 
-  // make any TupleUnpack dead by undoing TupleUnpack(TupleConstruct())
-  if(n->kind() == prim::TupleUnpack) {
-    auto construct = n->input()->node();
-    // note: removing these asserts changes this pass from a complete lowering
-    // pass to one that removes tuples when possible. When tuples are first-class
-    // in the interpreter, we should still run this pass to remove extraneous uses
-    JIT_ASSERTM(construct->kind() == prim::TupleConstruct, "tuple unpack not matched to tuple construct");
-    for(size_t i = 0; i < n->outputs().size(); ++i) {
-      n->outputs()[i]->replaceAllUsesWith(construct->inputs()[i]);
-    }
-    return;
-  }
+  // note: changing the second argument to false changes this pass from a complete lowering
+  // pass to one that removes tuples when possible. When tuples are first-class
+  // in the interpreter, we should still run this pass to remove extraneous uses
 
-  // undo TupleIndex[index=i](TupleConstruct())
-  if (n->kind() == prim::TupleIndex) {
-    auto construct = n->inputs().at(0)->node();
-    JIT_ASSERTM(construct->kind() == prim::TupleConstruct, "tuple index not matched to tuple construct");
-    auto idx = n->i(attr::index);
-    n->output()->replaceAllUsesWith(construct->inputs()[idx]);
-    return;
+  if (removeTupleNodes(n, /*must_remove_tuples*/true)) {
+    return; //tuple operator found
```

Comment:
Can we use:
```
if(n->kind() == TupleConstruct || TupleIndex || TupleSlice)
   removeTupleNodes(n, true);
   return
}
```
This more closely models the intent: that for these 3 nodes we handle them specially.
In the code as-is, you need to know that the combination of setting must_remove_tuples, and the return value ensures that you will hit the return.